exporter/azuredataexplorerexporter/adx_exporter.go

```diff
@@ -50,32 +56,46 @@ func (e *adxMetricsProducer) metricsDataPusher(ctx context.Context, metrics pmet
 		if err != nil {
 			e.logger.Error("Error performing serialization of data.", zap.Error(err))
 		}
-		ingestreader := bytes.NewReader(adxmetricjsonbytes)
-		if e.managedingest != nil {
-			if _, err := e.managedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
-				e.logger.Error("Error performing managed data ingestion.", zap.Error(err))
-				return err
-			}
-		} else {
-			if _, err := e.queuedingest.FromReader(context.Background(), ingestreader, e.ingestoptions...); err != nil {
-				e.logger.Error("Error performing queued data ingestion.", zap.Error(err))
-				return err
-			}
+		adxmetricjsonbytes = append(adxmetricjsonbytes, nextline...)
+		metricsByteArray = append(metricsByteArray, adxmetricjsonbytes...)
+	}
+	if len(metricsByteArray) != 0 {
+		if err := e.ingestData(metricsByteArray); err != nil {
+			return err
 		}
+		metricsByteArray = metricsByteArray[:0]
```

Comment:
This shows some linting error , what is the reason we are doing this.
![image](https://user-images.githubusercontent.com/106139410/174313468-e0303533-34be-4f6b-997f-28cbd7989b44.png)